src/main/index.js

```diff
@@ -604,6 +612,108 @@ function runApp() {
     }
   })
 
+  function manageTray(window, removeWindow = false) {
+    if (tray) {
+      if (!removeWindow) {
+        trayWindows.push(window)
+        createTrayContextMenu()
+      } else if (trayWindows.find(item => item.id === window.id)) {
+        trayClick(window)
+      }
+    } else {
+      const icon = process.env.NODE_ENV === 'development'
+        ? path.join(__dirname, '..', '..', '_icons', 'iconColor.png')
+        : path.join(__dirname, '..', '_icons', 'iconColor.png')
+
+      tray = new Tray(icon)
+
+      tray.setIgnoreDoubleClickEvents(true)
+      tray.setToolTip('FreeTube')
+
+      trayWindows = [window]
+      createTrayContextMenu()
+
+      if (process.platform !== 'linux') {
+        tray.on('click', (event) => {
+          if (trayWindows.length === 1) { trayClick(trayWindows[0]) }
+        })
+      }
+    }
+  }
+
+  function trayClick(window, close = false) {
+    if (!close) {
+      window.show()
+    } else if (trayWindows.length) {
+      window.destroy()
```

Comment:
Please use `window.close()` instead of `.destroy()`, as `.destroy()` bypasses all beforeunload handlers such as the one that saves the watch progress.